controllers/drain_controller.go

```diff
@@ -0,0 +1,458 @@
+		// TODO: make this time configurable
+		return &reconcile.Result{RequeueAfter: 5 * time.Second}, nil
+	}
+
+	if currentSnns == nil {
+		return nil, fmt.Errorf("failed to find sriov network node state for requested node")
+	}
+
+	err = utils.AnnotateObject(currentSnns, constants.NodeStateDrainAnnotationCurrent, constants.Draining, dr.Client)
+	if err != nil {
+		reqLogger.Error(err, "failed to annotate node with annotation", "annotation", constants.Draining)
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+func (dr *DrainReconcile) findNodePoolConfig(ctx context.Context, node *corev1.Node) (*sriovnetworkv1.SriovNetworkPoolConfig, []corev1.Node, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("findNodePoolConfig():")
+	// get all the sriov network pool configs
+	npcl := &sriovnetworkv1.SriovNetworkPoolConfigList{}
+	err := dr.List(ctx, npcl)
+	if err != nil {
+		logger.Error(err, "failed to list sriovNetworkPoolConfig")
+		return nil, nil, err
+	}
+
+	selectedNpcl := []*sriovnetworkv1.SriovNetworkPoolConfig{}
+
+	for _, npc := range npcl.Items {
+		// we skip hw offload objects
+		if npc.Spec.OvsHardwareOffloadConfig.Name != "" {
+			continue
+		}
+
+		// if the node selector is empty we skip
+		if npc.Spec.NodeSelector == nil {
+			continue
+		}
+
+		selector, err := metav1.LabelSelectorAsSelector(npc.Spec.NodeSelector)
+		if err != nil {
+			logger.Error(err, "failed to create label selector from nodeSelector", "nodeSelector", npc.Spec.NodeSelector)
+			return nil, nil, err
+		}
+
+		// If a pool with a nil or empty selector creeps in, it should match nothing, not everything.
+		if selector.Empty() || !selector.Matches(labels.Set(node.Labels)) {
+			continue
+		}
+
+		selectedNpcl = append(selectedNpcl, npc.DeepCopy())
+	}
+
+	if len(selectedNpcl) > 1 {
+		// don't allow the node to be part of multiple pools
+		err = fmt.Errorf("node is part of more then one pool")
+		logger.Error(err, "multiple pools founded for a specific node", "numberOfPools", len(selectedNpcl), "pools", selectedNpcl)
+		return nil, nil, err
+	} else if len(selectedNpcl) == 1 {
+		// found one pool for our node
+		logger.V(2).Info("found sriovNetworkPool", "pool", *selectedNpcl[0])
+		selector, err := metav1.LabelSelectorAsSelector(selectedNpcl[0].Spec.NodeSelector)
+		if err != nil {
+			logger.Error(err, "failed to create label selector from nodeSelector", "nodeSelector", selectedNpcl[0].Spec.NodeSelector)
+			return nil, nil, err
+		}
+
+		// list all the nodes that are also part of this pool and return them
+		nodeList := &corev1.NodeList{}
+		err = dr.List(ctx, nodeList, &client.ListOptions{LabelSelector: selector})
+		if err != nil {
+			logger.Error(err, "failed to list nodes using with label selector", "labelSelector", selector)
+			return nil, nil, err
+		}
+
+		return selectedNpcl[0], nodeList.Items, nil
+	} else {
+		logger.V(2).Info("using \"default\" sriovNetworkPool with MaxUnavailable of one", "pool", *defaultNpcl)
```

Comment:
Do we want to mention default pool in the log message?  Maybe we should change the message to say something like: node doesn't belong to any pool, using default drain configuration